components/jsx/preferences-modal/index.js

```diff
@@ -91,11 +91,26 @@ const getAlertsPreferences = ({ event, preferencesModal }) => {
 		}
 	});
 
-	const alertsEnabledText = preferencesList.innerHTML + addedTextBuffer.join(',') + '.';
-	const alertsDisabledText = 'You have disabled all instant alerts';
+	const alertsEnabledText = 'Your delivery channels: ' + addedTextBuffer.join(', ') + '.';
+	const alertsDisabledText = 'You have previously disabled all delivery channels';
 
 	preferencesList.innerHTML = addedTextBuffer.length > 0 ? alertsEnabledText : alertsDisabledText;
-}
+
+	try {
+		// We need the service worker registration to check for a subscription
+		const serviceWorkerRegistration = await navigator.serviceWorker.ready;
+		const subscription = await serviceWorkerRegistration.pushManager.getSubscription();
```

Comment:
*Change request*: I think we need to consider that the service worker is sometimes unavailable (e.g. the browser is in incognito mode)
```suggestion
		const serviceWorkerRegistration = await navigator.serviceWorker?.ready;
		const subscription = await serviceWorkerRegistration?.pushManager.getSubscription();
```

Reply:
That's a good point. However, in this case the error will be caught on line 102 and the browser option won't be added to the message. Also, it seems like the linter is showing warnings for optional chaining `?.`.
Let me know your thoughts.

Reply:
That's a good point. Since the error is caught properly, I think it is fine to keep it as is .